### ISSUE TITLE
Error message when running command "gradlew run"

### ISSUE THREAD

**Issue by ivan-marroquin** [igd-iva/colormap-explorer#19]
Hi there,

I am using a PC with windows 7. I get the following error message when I ran "gradlew run"

Copied natives into C:\Users\IMarroquin\Documents\My Research Work\Colores_work\Explorative_Analysis_2D_Colormaps\colormap-explorer-master\explorer/natives
:explorer:createVersionFile
Updated task ':explorer:createVersionFile' output files with 2018-07-16-08-34-34
:metrics:compileJava

FAILURE: Build failed with an exception.

* What went wrong:
Could not resolve all dependencies for configuration ':metrics:compile'.
> Could not resolve de.fhg.igd:pcolor:1.4.1.
  Required by:
      de.fhg.igd.iva:metrics:2018-07-16-08-34-33
   > Could not GET 'http://artifactory.esdi-humboldt.eu/artifactory/libs-release/de/fhg/igd/pcolor/1.4.1/pcolor-1.4.1.pom'.
      > Connection to http://artifactory.esdi-humboldt.eu refused

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 2 mins 2.285 secs

It seems that the installation can't download the file pcolor-1.4.1.pom

Any suggestions?

Many thanks,
Ivan

**Comment by msteiger** [comment]
Yes, looks like your assumption about the Artifactory server being down is correct. You can build or download the pcolor library here: https://github.com/igd-geo/pcolor

Maybe you want to raise an issue there wrt. uploading the library to JCenter or Maven Central.

**Comment by ivan-marroquin** [comment]
Hi Martin,

Thanks a lot for pointing the website to download pcolor. I was able to build the library and now it resides in this path:

Build complete.
The dist files are now in C:\Users\IMarroquin\Documents\My Research Work\Colores_work\Explorative_Analysis_2D_Colormaps\pcolor-master\de.fhg.igd.pcolor\build\libs

BUILD SUCCESSFUL

Could I ask you one more question? 

When I run the command "gradlew run" for colormap-explorer, what additional parameter do I need to provide in order to point to the location of pcolor?

Also, I would like to raise the issue to JCenter or Marven Central. I can't find the websites. By any chance do you know them?

Many thanks,
Ivan

**Comment by msteiger** [comment]
I think you can just run `gradlew publishToMavenLocal` on pcolor to install it into your local mvn repository.
Otherwise you can find the docs here: https://docs.gradle.org/current/userguide/publishing_maven.html

I suggest raising the issue here: https://github.com/igd-geo/pcolor/issues

**Comment by ivan-marroquin** [comment]
Thanks for the suggestion. I reported the incident to pcolor

**Comment by Epichide** [comment]
* setting pcolor jar for "gradlew run"


pcolor-1.4.1.jar ： [1.4.1.zip](https://github.com/user-attachments/files/17025609/1.4.1.zip)
copy jar to :  C:\Users\\[username]\\.m2\repository\com\example\pcolor\
add mavenLocal() to `build.geadle` file'  repositories block , then run `gradlew run`

```
subprojects {
...
	repositories {
		mavenLocal()
		mavenCentral()
...
```


* build "pcolor-1.4.1.jar "

>add "publishing" block in  `pcolor-master\build.gradle`  , and execute `gradlew publishToMavenLocal` , then pcolor-1.4.1.jar will be genrated

```
project(':de.fhg.igd.pcolor') {

  task packageJavadoc(type: Jar, dependsOn: 'javadoc') {
    from javadoc.destinationDir
    classifier = 'javadoc'
  }

  task packageSources(type: Jar, dependsOn: assemble) {
    from sourceSets.main.allSource
    classifier = 'sources'

    manifest {
      attributes['Bundle-Vendor'] = packageVendor
      attributes['Bundle-Description'] = packageDescription
      attributes['Eclipse-SourceBundle'] = project.name + ';version="' + project.version + '";roots:="."'
      attributes['Bundle-SymbolicName'] = project.name + ".source"
      attributes['Bundle-Version'] = project.version
    }
  }

// adding publishing
  publishing {
    publications {
      mavenJava(MavenPublication) {
            groupId = 'de.fhg.igd' 
            artifactId = 'pcolor' 
            version = project.version 
        from components.java
        artifact packageJavadoc
        artifact packageSources
      }
    }
    repositories {
      maven {
        name = 'localRepo'
        url = uri("${project.buildDir}/repo")
      }
    }
  }

  // include doc and source in build
  tasks.build.dependsOn += [packageSources, packageJavadoc]
}
```
